### PR TITLE
 feat(ui5-product-switch): initial implementation

### PR DESCRIPTION
**Pull request by nnaydenow**
FIXES: #857

### CODE REVIEW

File: packages/fiori/src/ProductSwitch.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			type: HTMLElement,
+		},
+	},
+};
+
+/**
+ * @class
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-product-switch</code> is a Fiori specific control that is used in <code>ui5-shellbar</code>
```
Comment (vladitasev) [inline packages/fiori/src/ProductSwitch.js]: wrong word here

File: packages/fiori/src/ProductSwitch.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ * <br><br>
+ * <h3>ES6 Module Import</h3>
+ * <code>import "@ui5/webcomponents-fiori/dist/ProductSwitch";</code>
```
Comment (vladitasev) [inline packages/fiori/src/ProductSwitch.js]: better end the import with .js

File: packages/fiori/src/ProductSwitch.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.ProductSwitch
```
Comment (vladitasev) [inline packages/fiori/src/ProductSwitch.js]: fiori, not main

File: packages/fiori/src/ProductSwitch.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	}
+
+	static get template() {
+		return ProductSwitchTemplate;
+	}
+
+	onEnterDOM() {
+		this._handleResizeBound = this._handleResize.bind(this);
+
+		ResizeHandler.register(document.body, this._handleResizeBound);
```
Comment (vladitasev) [inline packages/fiori/src/ProductSwitch.js]: why on the body?

File: packages/fiori/src/ProductSwitch.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+ * @tagname ui5-product-switch
+ * @appenddocs ProductSwitchItem
+ * @public
+ * @since 1.0.0-rc.5
+ */
+class ProductSwitch extends UI5Element {
+	constructor() {
+		super();
+
+		this.ROW_MIN_WIDTH = {
```
Comment (vladitasev) [inline packages/fiori/src/ProductSwitch.js]: This isn't per component, should be on the constructor function

File: packages/fiori/src/ProductSwitchItem.js [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+		subtitle: {
+			type: String,
+		},
+		/**
+		 * Defines the icon to be displayed as a graphical element within the <code>ui5-product-switch-item</code>.
+		 * The SAP-icons font provides numerous options.
+		 * <br><br>
+		 * Example:
+		 * <br>
+		 * <pre>ui5-product-switch-item icon="sap-icon://palette"</pre>
```
Comment (vladitasev) [inline packages/fiori/src/ProductSwitchItem.js]: sap-icon:// should not be written by users any more, we deprecated it

File: packages/fiori/src/ProductSwitchItem.js [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+ * <h3 class="comment-api-title">Overview</h3>
+ * The <code>ui5-product-switch-item</code> represents the items displayed in the <code>ui5-product-switch</code> web component
+ * <b>Note:</b> <code>ui5-product-switch-item</code> is not supported when used outside of <code>ui5-product-switch</code>.
+ * <br><br>
+ * <h3>ES6 Module Import</h3>
+ * <code>import "@ui5/webcomponents-fiori/dist/ProductSwitchItem";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.ProductSwitchItem
```
Comment (vladitasev) [inline packages/fiori/src/ProductSwitchItem.js]: fiori, not main.

Overall: please scan all the files for the word "main" to make sure there aren't any other leftovers

File: packages/fiori/src/themes/ProductSwitch.css [hunk 8]
```diff
@@ -0,0 +1,30 @@
+:host {
+	font-family: var(--sapUiFontFamily);
```
Comment (ilhan007) [inline packages/fiori/src/themes/ProductSwitch.css]: Usually,  we write css regarding the html native 'hidden' attribute, so the user can hide our components, using it.

```css 
!-- from Button.css --!
:host(:not([hidden])) {
	display: inline-block;
}
```

File: packages/fiori/src/ProductSwitchItem.js [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+		 */
+		click: {},
+		_focused: {},
+	},
+};
+
+/**
+ * @class
+ * <h3 class="comment-api-title">Overview</h3>
+ * The <code>ui5-product-switch-item</code> represents the items displayed in the <code>ui5-product-switch</code> web component
```
Comment (xtatica) [inline packages/fiori/src/ProductSwitchItem.js]: missing a full stop at the end

File: packages/fiori/src/ProductSwitch.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+			type: HTMLElement,
+		},
+	},
+};
+
+/**
+ * @class
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-product-switch</code> is a Fiori specific web component that is used in <code>ui5-shellbar</code> and allows the user easy switching between products.
```
Comment (xtatica) [inline packages/fiori/src/ProductSwitch.js]: The <code>ui5-product-switch</code> is a**n SAP** Fiori specific web component that is used in <code>ui5-shellbar</code> and allows the user **to easily switch** between products.

**Pull request merged**